plugins/inventory/nb_inventory.py

```diff
@@ -820,6 +856,32 @@ def get_region_for_site(site):
         # Dictionary of site id to region id
         self.sites_region_lookup = dict(map(get_region_for_site, sites))
 
+    # Note: depends on the result of refresh_sites_lookup for self.sites_with_prefixes
+    def refresh_prefixes(self):
+        # Pull all prefixes defined in NetBox
+        url = self.api_endpoint + "/api/ipam/prefixes"
+
+        if self.fetch_all:
+            prefixes = self.get_resource_list(url)
+        else:
+            prefixes = self.get_resource_list_chunked(
+                api_url=url,
+                query_key="site",
+                query_values=list(self.sites_with_prefixes),
+            )
+        self.prefixes_sites_lookup = defaultdict(dict)
+        self.prefixes_lookup = defaultdict(dict)
+
+        # We are only concerned with Prefixes that have actually been assigned to sites
+        for prefix in prefixes:
+            if prefix.get("site"):
+                prefix_id = prefix["id"]
+                site_id = prefix["site"]["id"]
+                self.prefixes_lookup[prefix_id] = prefix
+                self.prefixes_sites_lookup[site_id] = prefix_id
+            # Remove "site" attribute, as it's redundant when prefixes are assigned to site
```

Comment:
This may still be useful to trace back to a site, if the prefix ends up getting shared with another function or something else. I don't think it's wise to delete expected attributes

Reply:
This was done to follow the precedent set by the refresh_ipaddresses() function.
 In refresh_ipaddresses() the "interface" attribute is removed because this information is redundant once the whole IP Address dictionary is nested within self.ipaddresses_intf_lookup[interface_id][ip_id] and self.ipaddresses_lookup[ip_id].

I am fine with leaving this attribute, but removing it provides the user with a cleaner inventory. (less redundant info)

Reply:
Ok, if it folllows an existing convention then that's fine